apis/resourcecontrollerv2/v1alpha1/resource_instance_types.go

```diff
@@ -68,74 +64,80 @@ type ResourceInstanceParameters struct {
 // ResourceInstanceObservation are the observable fields of a ResourceInstance.
 type ResourceInstanceObservation struct {
 	// The ID associated with the instance.
-	ID string `json:"id,omitempty"`
+	ID *string `json:"id,omitempty"`
```

Comment:
We typically don't use pointers in status as we usually aren't concerned with whether the fields is zero-valued or nil, but I'll let you make a call here :)

Reply:
Good point. The [IBM Platform Services SDK](https://github.com/IBM/platform-services-go-sdk), which is also being code generated from OpenAPI v3 specs, is using pointers in pretty much all go types. But for consistency, it seems like a good rule of thumb could be that:
1. In spec mandatory fields should be by value, and optional fields pointers
2. In status, all fields should be by value
Does it make sense ?

Reply:
Yep that is generally the pattern we follow :) Here is the relevant design doc: https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers